openapi/generator.go

```diff
@@ -370,12 +377,25 @@ func (g *Generator) setOperationResponse(op *Operation, t reflect.Type, code, mt
 		Content:     make(map[string]*MediaTypeOrRef),
 		Headers:     make(map[string]*HeaderOrRef),
 	}
+
+	// cast examples if given
+	var castedExamples map[string]*ExampleOrRef
+	if examples != nil {
+		castedExamples = make(map[string]*ExampleOrRef)
+		for name, val := range examples {
+			castedExamples[name] = &ExampleOrRef{Example: &Example{Value: val}}
+		}
+	}
+
 	// The response may have no content type specified,
 	// in which case we don't assign a schema.
 	schema := g.newSchemaFromType(t)
-	if schema != nil {
+
+	if schema != nil || example != nil || castedExamples != nil {
```

Comment:
Unsure about that. The presence of the schema is necessary, otherwise the examples represent nothing. 

Since we're already checking that `example` and `examples` are not both set, we could simply assign `example` and `castedExamples` (that will be nil if `examples` is) and be done with it.

Reply:
I could'nt find confirmation in the spec that a `MediaType` object MUST have a schema, so I guess you are right, the schema isn't mandatory.